examples/minimal/packages/client-react-external-wallet/src/mud/systemCalls.ts

```diff
@@ -0,0 +1,15 @@
+import { type WalletClient } from "viem";
+import { type MUDNetwork } from "./NetworkContext";
+import IWorldAbi from "contracts/out/IWorld.sol/IWorld.abi.json";
+
+export const increment = async (client: WalletClient, network: MUDNetwork) => {
+  const { request } = await network.publicClient.simulateContract({
+    address: network.worldAddress,
+    abi: IWorldAbi,
+    functionName: "increment",
+    args: [],
+    account: client.account,
+  });
+  const tx = await client.writeContract(request);
+  network.waitForTransaction(tx);
```

Comment:
would be neat if this was a custom client action too, which means we could simplify the args to potentially just the public client + wallet client

there's a bit of a chicken-and-egg problem though, where functions like `syncToZustand` require the public client and return `waitForTransaction`, which we'd need to extend the public client

maybe we just pass in the plain public client to set up the sync then extend it before returning it to the rest of the app?

```ts
const publicClient = createPublicClient(...);
const syncResult = syncToZustand({ publicClient, ... }) 

return {
  publicClient: publicClient.extend(syncActions(syncResult)),
  ...
}
```

thoughts?

Reply:
Having waitForTransaction as a view client action makes sense. I'll look into this.